src/org/opensolaris/opengrok/web/PageConfig.java

```diff
@@ -391,8 +392,22 @@ public String canProcess() {
                 dirFileList = Collections.emptyList();
             } else {
                 Arrays.sort(files, String.CASE_INSENSITIVE_ORDER);
-                dirFileList
-                        = Collections.unmodifiableList(Arrays.asList(files));
+                List<String> list = Arrays.asList(files);
+                if (getPath().isEmpty()) {
+                    /**
+                     * This denotes the source root directory, we need to filter
+                     * projects which aren't allowed by the authorization. E. g.
```

Comment:
adjust the comment explaining what method called below actually performs the authorization check